pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1007,27 +1007,45 @@ func (p *PriorityQueue) Update(logger klog.Logger, oldPod, newPod *v1.Pod) error
 	if usPodInfo := p.unschedulablePods.get(newPod); usPodInfo != nil {
 		pInfo := updatePod(usPodInfo, newPod)
 		p.updateNominatedPodUnlocked(logger, oldPod, pInfo.PodInfo)
+		gated := usPodInfo.Gated
+		if p.isSchedulingQueueHintEnabled {
+			// When unscheduled Pods are updated, we check with QueueingHint
+			// whether the update may make the pods schedulable.
+			// Plugins have to implement a QueueingHint for Pod/Update event
+			// if the rejection from them could be resolved by updating unscheduled Pods itself.
+			hint := p.isPodWorthRequeuing(logger, pInfo, UnscheduledPodUpdate, oldPod, newPod)
```

Comment:
Worth to keep an eye on the performance tests for massive gated pods. This might be a minimum comparison for queueingHints vs non-queueingHints because the logic for schedulingGate plugin is simple.